### PR TITLE
Feature/upload images

### PR DESCRIPTION
**Pull request by JeanVittory**
# SUMMARY

Created a middleware to upload images using cloudinary.

## TO-DO
start implementing what is necessary to update users and provide products to the frontend as well as start to  build the appointment process in backend

### CODE REVIEW

File: src/api/middlewares/formData.middlewares.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { Busboy } from 'busboy';
+import { Stream } from 'stream';
+import { UploadStream } from 'cloudinary';
+import { Request, Response, NextFunction } from 'express';
+import { ApiError } from '../../config/middlewares/errorHandler/ApiError.middlewares';
+import { OBJ_FORM_DATA } from '../utils/constants.utils';
+
+const formData = (folderName: string) => {
+  return (req: Request, res: Response, next: NextFunction): void => {
+    const bb: Busboy = busboy({ headers: req.headers });
```
Comment (ht1204) [inline src/api/middlewares/formData.middlewares.ts:12]:
```js
const { headers } = req;
const busboy_stream: Busboy = busboy({ headers });
```

File: src/api/middlewares/formData.middlewares.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    let uploadingCount: number = 0;
+    let uploadingFile: boolean = false;
+
+    const done = (): void => {
+      if (uploadingFile) return;
+      if (uploadingCount > 0) return;
```
Comment (ht1204) [inline src/api/middlewares/formData.middlewares.ts:18]: `if (uploadingCount) return;`

File: src/api/middlewares/formData.middlewares.ts (hunk 1 above)
Comment (JeanVittory) [reply src/api/middlewares/formData.middlewares.ts:12]: DONE!

File: src/api/middlewares/formData.middlewares.ts (hunk 2 above)
Comment (JeanVittory) [reply src/api/middlewares/formData.middlewares.ts:18]: DONE!

**Review by ht1204 — APPROVED** [review]

**Pull request merged**